### PR TITLE
Side-by-side plot of user selected variables across scenarios

### PR DESCRIPTION
**Pull request by tonnrueter**
Function generates a facet wrap/grid plot that shows user defined variables side by side for comparison across different scenarios.  Solves [Improve cs2 climate plots](https://github.com/pik-piam/remind2/issues/487) issue. Plot examples can be found in [here](https://gitlab.pik-potsdam.de/tonnru/mip_dev/-/blob/main/tests_plotPercentiles.nb.html)

### CODE REVIEW

**Review by fbenke-pik — CHANGES_REQUESTED** [review]
Please add the man page for your newly created function as well.

**Comment by fbenke-pik** [comment]
Currently, using the `variables` option is not working as expected

```
data <- readxl::read_xlsx("~/Downloads/ngfs-magicc-data.xlsx")
vars <- c(
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|10.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|16.7th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|17.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|25.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|33.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|5.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|50.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|66.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|67.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|75.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|83.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|83.3th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|90.0th Percentile",
  "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|95.0th Percentile"
)

df <- quitte::as.quitte(data)
mip::plotPercentiles(df, variables = vars)
```
This gives the error

`Warning: 'AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|10.0th Percentile' missing in available dataError in mip::plotPercentiles(df, variables = vars) : 
  Provided variable is missing in data`

File: R/plotPercentiles.R
Comment (fbenke-pik) [inline R/plotPercentiles.R:1]: This is not a good description of the function, as it basically applies to any plot function in this library.

File: R/plotPercentiles.R
Comment (fbenke-pik) [inline R/plotPercentiles.R:18]: Assumptions about the structure of the data frame should be moved to the description of the parameter `df` in the function documentation so they can be viewed when calling `help("plotIterations")`. User's should not have to look into the code to find out what the structure of the input data is.

**Comment by tonnrueter** [comment]
> Currently, using the `variables` option is not working as expected
> 
> ```
> data <- readxl::read_xlsx("~/Downloads/ngfs-magicc-data.xlsx")
> vars <- c(
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|10.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|16.7th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|17.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|25.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|33.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|5.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|50.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|66.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|67.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|75.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|83.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|83.3th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|90.0th Percentile",
>   "AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|95.0th Percentile"
> )
> 
> df <- quitte::as.quitte(data)
> mip::plotPercentiles(df, variables = vars)
> ```
> 
> This gives the error
> 
> `Warning: 'AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3|10.0th Percentile' missing in available dataError in mip::plotPercentiles(df, variables = vars) : Provided variable is missing in data`

Try `AR6 climate diagnostics|Net Atmosphere to Land Flux|CO2|MAGICCv7.5.3`, ie. the variable prefix without the Percentile suffix

File: R/plotPercentiles.R
Comment (fbenke-pik) [inline R/plotPercentiles.R:33]: The following part can be drastically simplified by writing
`if(!all(scenarios %in% unique(data$scenario))) {...}`

Same goes for variables.

Comment (fbenke-pik) [reply R/plotPercentiles.R:33]: If you want to list missing variables / scenarios explicitly in a warning, you can use `setdiff(unique(data$scenario), scenarios)` .  For nice printing, you can use 
`paste0(setdiff(unique(data$scenario), scenarios), collapse = ", ")`

File: R/plotPercentiles.R
Comment (fbenke-pik) [inline R/plotPercentiles.R:62]: This is another assumption about the data you must state explicitly in the documentation. As of now, you expect variables for 5,33,50,67,95 the percentile with very specific names in the input data, so add it to the documentation.

(In the future, we might want to make this more flexible. But this will be for the stakeholders - Oliver and Gabriel - to decide)

**Comment by fbenke-pik** [comment]
https://github.com/pik-piam/mip/pull/84#issuecomment-1880700584

Ok, so please enhance the description of the `variables` parameter, so that it is clear what is expected (a list of variables without the percentile fragments)

File: R/plotPercentiles.R
Comment (tonnrueter) [reply R/plotPercentiles.R:33]: I don't think we can, since we have three possible outcomes:
1. User does not provide a list of scenarios (scenarios == NULL) -> Use all available scenarios
2. User provides list containing a scenario not present in data -> Throw error
3. User provides list of valid scenarios -> Use those

Helper function `allItemsAvailable` also lets the user know *which* scenario is missing from data

Comment (fbenke-pik) [reply R/plotPercentiles.R:33]: Just add a pre-check for outcome 1 (an if-clause), and use setdiff check to determine if outcome 2 or 3 applies otherwise. Also, I added a recommendation for showing warnings above (a combo of `setdiff` and `paste0`).

File: R/plotPercentiles.R
Comment (tonnrueter) [reply R/plotPercentiles.R]: Fixed

File: R/plotPercentiles.R
Comment (tonnrueter) [reply R/plotPercentiles.R:39]: Fixed

File: R/plotPercentiles.R
Comment (fbenke-pik) [inline R/plotPercentiles.R:2]: Here you write "if available" (as in: optional), below you say "must be given" (as in: mandatory).  Can you clarify?

File: R/plotPercentiles.R
Comment (fbenke-pik) [inline R/plotPercentiles.R:8]: " In the quitte data frame all  percentiles must be given as individual variables ..."
Please add an example to avoid misunderstandings.

"Manipulate input data frame ....." 
This should not be here, as it is code documentation, but not a specification of the input parameter and therefore nothing a user has to worry about, right?

**Review by fbenke-pik — APPROVED** [review]
Implementation looks great, thank you! Now, we have to wait for @orichters approval to be sure that the function works as intended by him.

**Review by orichters — APPROVED** [review]
Thanks a lot, I looked at [the examples](https://gitlab.pik-potsdam.de/tonnru/mip_dev/-/blob/main/tests_plotPercentiles.nb.html) and they look really good, as I hoped for. We could now add that to [the climate section of compareScenarios2](https://github.com/pik-piam/remind2/blob/master/inst/markdown/compareScenarios2/cs2_07_climate.Rmd) (conditional on data availability)? I did not check the code, but trust Falk's assessment here

**Comment by fbenke-pik** [comment]
> We could now add that to [the climate section of compareScenarios2](https://github.com/pik-piam/remind2/blob/master/inst/markdown/compareScenarios2/cs2_07_climate.Rmd) (conditional on data availability)? 

I think this needs to be merged first? https://github.com/remindmodel/remind/pull/1475
Anyways, as soon as you can give us a run that contains these variables, we can extend remind2.

> I did not check the code, but trust Falk's assessment here
The code is fine, but some decisions concerning the expected format of the input parameters are debatable (as in: there is no right or wrong given the data we have right now). We probably have to revisit this point once we have a run with actual data.

**Pull request merged**
